src/binary-reader-ir.cc

```diff
@@ -818,8 +786,7 @@ Result BinaryReaderIR::BeginDataSegment(Index index, Index memory_index) {
   auto field = MakeUnique<DataSegmentModuleField>(GetLocation());
   DataSegment& data_segment = field->data_segment;
   data_segment.memory_var = Var(memory_index, GetLocation());
-  module->data_segments.push_back(&data_segment);
-  module->fields.push_back(field.release());
+  module->AppendField(field.release());
```

Comment:
What are some of these .release() and others std::move()?  (sorry basic C++ question).

Reply:
`.release()` is used when the caller wants a raw pointer and `std::move()` is used when the caller wants a `std::unique_ptr`. It probably should be a `unique_ptr` in all cases, actually, since they're always taking ownership.

Reply:
PR to do this: #602.